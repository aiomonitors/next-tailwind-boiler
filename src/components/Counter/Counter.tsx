import React, {useState} from 'react';
import styled from 'styled-components';

export type CounterProps = {
	initialValue?: number;
};

export const Counter = ({initialValue = 0}: CounterProps) => {
	const [count, setCount] = useState(initialValue);

	const onIncrement = () => {
		setCount((prevCount) => prevCount + 1);
	};

	const onReset = () => {
		setCount(initialValue);
	};

	return (
		<CounterWrapper>
			<span>Current count is: {count}</span>
			<ButtonsWrapper>
				<CounterButton onClick={onIncrement}>Increment</CounterButton>
				<CounterButton onClick={onReset}>Reset</CounterButton>
			</ButtonsWrapper>
		</CounterWrapper>
	);
};

const CounterWrapper = styled.div`
    display: flex;
    flex-direction: column;
    gap: 10px;
`;

const ButtonsWrapper = styled.div`
    display: flex;
    justify-content: space-evenly;
`;

const CounterButton = styled.button`
  padding: 10px 15px;
  color: rgba(255, 255, 255, 0.5);
  font-size: 1.1rem;
  font-weight: 500;
  background-color: transparent;
  outline: none;
  border: 1px solid white;
  border-radius: 4px;
  transition: all 0.2s ease-in-out;

  &:hover {
    cursor: pointer;
    color: rgba(255, 255, 255, 1);
  }
`;
