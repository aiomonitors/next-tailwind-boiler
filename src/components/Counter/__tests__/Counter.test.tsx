import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Counter} from '../Counter';

Enzyme.configure({adapter: new Adapter()});

describe('Counter', () => {
	test('It matches the snapshot', () => {
		const counter = shallow(<Counter />);
		expect(counter).toMatchSnapshot();
	});

	test('It should have a default value of 0', () => {
		const counter = shallow(<Counter />);
		expect(counter.find('span').text()).toEqual('Current count is: 0');
	});

	test('It should set default value to initialValue passed in', () => {
		const counter = shallow(<Counter initialValue={5} />);
		expect(counter.find('span').text()).toEqual('Current count is: 5');
	});
    
	test('It should increment the value when Increment is pressed', () => {
		const counter = mount(<Counter initialValue={5} />);
		counter.find('button').at(0).simulate('click');
		expect(counter.find('span').text()).toEqual('Current count is: 6');
	});

	test('It should reset the value when Reset is pressed', () => {
		const counter = mount(<Counter initialValue={5} />);
		counter.find('button').at(0).simulate('click');
		expect(counter.find('span').text()).toEqual('Current count is: 6');
		counter.find('button').at(1).simulate('click');
		expect(counter.find('span').text()).toEqual('Current count is: 5');
	});
});