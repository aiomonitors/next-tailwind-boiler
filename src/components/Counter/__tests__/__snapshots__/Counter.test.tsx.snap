// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter It matches the snapshot 1`] = `ShallowWrapper {}`;
